### PR TITLE
Minor fixes

### PR DESCRIPTION
**Pull request by DrMarcII**
Prevent multiple in-flight requests to WebDriver (WebDriver is not thread-safe).
Fix type issue when getting cookies back in some instances.
Add waitFor to refresh and maximize tests to prevent failures under some circumstances.

### CODE REVIEW

**Comment by DrMarcII** [comment]
@devoncarew 
A collection of minor fixes. Can you check if the changes to the tests get them passing on your end?

**Comment by devoncarew** [comment]
I ran it twice and got consistent results:

```
FAIL: Window maximize
  Expected: a value less than <100>
    Actual: <100>
     Which: is not a value less than <100>
  
  test/src/window_test.dart 1:1  main.<fn>.<fn>.<<fn>_async_body>
  dart:isolate                   _RawReceivePortImpl._handleMessage
  


64 PASSED, 1 FAILED, 0 ERRORS
```

One test did take quite a while to run - not sure which one, and if it timed out?

File: lib/src/command_processor.dart [hunk 1]
```diff
@@ -59,4 +64,19 @@ class _CommandProcessor {
     request.headers.add(HttpHeaders.ACCEPT_CHARSET, UTF8.name);
     request.headers.add(HttpHeaders.CACHE_CONTROL, "no-cache");
   }
+
+  Future _getLock() async {
```
Comment (devoncarew) [inline lib/src/command_processor.dart]: This locking mechanism is great! Perhaps move it into it's own class (`Lock`) to make it re-usable?

Comment (DrMarcII) [inline lib/src/command_processor.dart]: Done

**Comment by DrMarcII** [comment]
@devoncarew 
The maximize test would take a minute to fail when it fails. I have dropped the default timeout for waitFor to 5 seconds so it will fail faster.

If you want to change the Window maximize test to pass on OSX while still testing something useful, feel free. It has been a while since I have played with a Mac, but it might be as easy as just removing the expect(location.x...) line and to change the waitFor to look for one of the other properties to change.

**Comment by devoncarew** [comment]
I sent you a PR against your master branch, which fixes the mac maximize test for me -

**Pull request merged**